kmath-trajectory/src/commonMain/kotlin/space/kscience/kmath/trajectory/segments/components/Pose2D.kt

```diff
@@ -0,0 +1,13 @@
+package space.kscience.kmath.trajectory.segments.components
+
+import space.kscience.kmath.geometry.Vector2D
+import kotlin.math.cos
+import kotlin.math.sin
+
+public data class Pose2D(
+    override val x: Double,
+    override val y: Double,
+    public val theta: Double
+) : Vector2D {
+    internal constructor(vector: Vector2D, theta: Double) : this(vector.x, vector.y, theta)
```

Comment:
I prefer factory functions to secondary constructors

Reply:
Fixed